python/modules/qcdSmearProducer.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python
+import os, sys
+import ROOT
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+from importlib import import_module
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection, Object
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoAODTools.postprocessing.tools import deltaPhi, deltaR, closest
+
+class qcdSmearProducer(Module): 
+    def __init__(self):
+        self.writeHistFile=True
+        self.metBranchName="MET"
+        self.xBinWidth = 0.01
+        self.minWindow = 0.01
+        self.maxWindow = 0.5
+        self.nSmears = 100
+        self.nSmearJets = 2
+        self.nBootstraps = 50
+        self.LINEAR_GRANULATED=True
+        self.winType = self.LINEAR_GRANULATED
+        self.doFlatSampling = True
+        self.canSmear  =True
+        self.respInputName = "JetResByFlav"
+        self.respFileName = "file:/eos/uscms/store/user/ddash/qcd_smeared/resTailOut_combined_filtered_CHEF_puWeight_weight_WoH_NORMALIZED.root"
```

Comment:
How is this file path will be used? I think calling file from EOS should via xrootd. And if this file will be called by many jobs, you need to copy it locally. Otherwise it will crash the EOS.

Reply:
Yes, it should be saved locally. Finally, I will change it.